frontend/appflowy_flutter/lib/plugins/database_view/grid/application/row/row_bloc.dart

```diff
@@ -106,16 +108,19 @@ class RowState with _$RowState {
 
   factory RowState.initial(
     CellContextByFieldId cellByFieldId,
-  ) =>
-      RowState(
-        cellByFieldId: cellByFieldId,
-        cells: UnmodifiableListView(
-          cellByFieldId.values
-              .map((e) => GridCellEquatable(e.fieldInfo))
-              .toList(),
-        ),
-        rowSource: const RowSourece.disk(),
-      );
+  ) {
+    cellByFieldId.removeWhere((_, cellContext) => !cellContext.isVisible());
+    return RowState(
+      cellByFieldId: cellByFieldId,
+      cells: UnmodifiableListView(
+        cellByFieldId.values
+            .where((cellContext) => cellContext.isVisible())
+            .map((e) => GridCellEquatable(e.fieldInfo))
+            .toList(),
```

Comment:
You have to modifiers that achieve the same here I believe, the `removeWhere` and the `where`.

Seems like you can make do with only one.